packages/back-end/src/controllers/organizations.ts

```diff
@@ -452,6 +452,47 @@ export async function putMemberRole(
   }
 }
 
+export async function putInviteRole(
+  req: AuthRequest<{ role: MemberRole }, { key: string }>,
+  res: Response
+) {
+  req.checkPermissions("organizationSettings");
+
+  const { org } = getOrgFromReq(req);
+  const { role } = req.body;
+  const { key } = req.params;
+
+  let found = false;
+
+  org.invites.forEach((m) => {
+    if (m.key === key) {
+      m.role = role;
+      found = true;
+    }
+  });
+
+  if (!found) {
+    return res.status(404).json({
+      status: 404,
+      message: "Cannot find member",
+    });
+  }
+
+  try {
+    await updateOrganization(org.id, {
+      invites: org.invites,
+    });
```

Comment:
We should add an audit log entry here so we can keep track of who made this change.

It requires keeping a copy of the original `invites` array and then doing this after updating:

```ts
    await req.audit({
      event: "organization.update",
      entity: {
        object: "organization",
        id: org.id,
      },
      details: auditDetailsUpdate(
        {invites: originalInvites}, // Starting value
        {invites: org.invites}      // Ending value
      )
    });
```

Reply:
@jdorn , Thanks for the comment. Audit log added